keel-sql/src/main/kotlin/com/netflix/spinnaker/keel/sql/SqlResourceRepository.kt

```diff
@@ -114,34 +113,14 @@ class SqlResourceRepository(
         field("spec") to objectMapper.writeValueAsString(resource.spec)
       )
       val insertPairs = updatePairs + (field("uid") to uid)
-      try {
-        insertInto(
-          RESOURCE,
-          *insertPairs.keys.toTypedArray()
-        )
-          .values(insertPairs.values)
-          .onDuplicateKeyUpdate()
-          .set(updatePairs)
-          .execute()
-      } catch (e: SQLDialectNotSupportedException) {
```

Comment:
Newer JOOQ supports `onDuplicateKeyUpdate` with H2. No need for this fallback any more.